stories/__tests__/SQFormGuidedWorkflow.stories.test.js

```diff
@@ -0,0 +1,249 @@
+      name: /toggle-expansion/i,
+      expanded: false
+    });
+    expect(remainingSections.length).toBe(2);
+
+    const agentScript = screen.getByRole('heading', {name: /agent script/i});
+    expect(agentScript).toBeVisible();
+
+    //Should we also check for the icon and script text?
+
+    const outcomeArea = screen.getByRole('heading', {name: /confirm info/i});
+    expect(outcomeArea).toBeVisible();
+
+    //Should we also check for the dropdown and textarea?
+
+    const resetButton = screen.getByRole('button', {name: /reset form/i});
+    expect(resetButton).toBeVisible();
+    expect(resetButton).toBeDisabled();
+
+    const nextButton = screen.getByRole('button', {name: /form submission/i});
+    expect(nextButton).toBeVisible();
+    expect(nextButton).toBeDisabled();
+  });
+
+  it('should enable next button when all required fields are filled out in the 1st section', () => {
+    render(<SQFormGuidedWorkflow />);
+
+    const nextButton = screen.getByRole('button', {name: /form submission/i});
+    expect(nextButton).toBeDisabled();
+
+    const outcome = screen.getByRole('button', {name: /outcome/i});
+    userEvent.click(outcome);
+
+    const interested = screen.getByText('Interested');
+    userEvent.click(interested);
+
+    expect(nextButton).toBeEnabled();
+  });
+
+  it('should close curent expanded card and open the next one when next button is clicked', async () => {
+    render(<SQFormGuidedWorkflow />);
+
+    const introText = screen.queryByText(/hi, bob smith, my name is/i);
+    expect(introText).toBeVisible();
+
+    const policyText = screen.queryByText(/stuff about policy cancellation/i);
+    expect(policyText).not.toBeVisible();
+
+    const outcome = screen.getByRole('button', {name: /outcome/i});
+    userEvent.click(outcome);
+
+    const interested = screen.getByText('Interested');
+    userEvent.click(interested);
+
+    const nextButton = screen.getByRole('button', {name: /form submission/i});
+    userEvent.click(nextButton);
+
+    await waitFor(
+      () =>
+        expect(
+          screen.getByText(/stuff about policy cancellation/i)
+        ).toBeVisible(),
+      {timeout: 5000}
+    );
+
+    expect(introText).not.toBeVisible();
+    expect(screen.getByText(/stuff about policy cancellation/i)).toBeVisible();
+  });
+
+  it("should allow user to go reopen the 1st section after it's completed", async () => {
+    //This test runs long so extending the time limit
+    jest.setTimeout(30000);
+    render(<SQFormGuidedWorkflow />);
+
+    const introText = screen.getByText(/hi, bob smith, my name is/i);
+    expect(introText).toBeVisible();
+
+    const outcome = screen.getByRole('button', {name: /outcome/i});
+    userEvent.click(outcome);
+
+    const interested = screen.getByText('Interested');
+    userEvent.click(interested);
+
+    const nextButton = screen.getByRole('button', {name: /form submission/i});
+    userEvent.click(nextButton);
+
+    await waitFor(
```

Comment:
I feel like these `waitFor`s should be changed to `await findByText`s, but it seems like the timeout isn't working properly with that...
I'm still not very familiar with this component, but maybe some `await`ing for elements to be visible/not visible between those button clicks might help with the timing?

Reply:
This line does have a wait for visible. I'm not sure I understand your comment. I can try using findByText instead. It's been a while so I can't remember if I tried that or not.
The big reason for the wait here is that part of the story includes a 3 second timer to demonstrate using a loading spinner for this particular model. And also the transitions between modules usually needs a wait of some kind.

Reply:
Using `await findByText` doesn't seem to be working ☹️

Reply:
I dug into this a little more and it seems like `findByText` is immediately returning since the text does exist in the DOM. However, it's not visible, which is what we want to check for. `waitFor` allows me to wait for visibility.